plugins/modules/zabbix_host.py

```diff
@@ -1085,9 +1084,9 @@ def main():
                     macro.pop('type')
                 else:
                     if macro['type'] == 'text':
-                        macro['type'] = 0
+                        macro['type'] = '0'
```

Comment:
Would this break some evaluations? For example one on line 812?

Reply:
We need to remove the whole block 809-813 as we don't want to `macro.pop('value', False)` if we follow the logic "secret macro value will have to be updated always" (see my comment above please)

Reply:
Ok, I guess it never being called is also a way to achieve that :) Feel free to cleanup in a follow up PR if you feel like it

Reply:
Submitted https://github.com/ansible-collections/community.zabbix/pull/724